src/main/java/com/zendesk/maxwell/producer/MaxwellKafkaProducer.java

```diff
@@ -129,9 +142,11 @@ public void push(RowMap r) throws Exception {
 
 		ProducerRecord<String, String> record;
 		if (r instanceof DDLMap) {
-			record = new ProducerRecord<>(ddlTopic, this.ddlPartitioner.kafkaPartition(r, this.numPartitions), key, value);
+			String topic = generateTopic(this.ddlTopic, r);
+			record = new ProducerRecord<>(topic, this.ddlPartitioner.kafkaPartition(r, getNumPartitions(topic)), key, value);
```

Comment:
if we're gonna support this feature for the DDL topic (and I'm honestly not sure we should), we need to rework `DDLMap#getDatabase()` and `DDLMap#getTable()` -- at the moment it looks as though those are currently just hard-coded to some defaults, rather than fished out of the `ResolvedSchemaChange` object